### PR TITLE
Added a new "symfony demo" command

### PR DESCRIPTION
**Pull request by javiereguiluz**
This is a WIP and early release of the new `symfony demo` command. The general idea around this command is the following:

  * Soon the `AcmeDemoBundle` will be gone forever.
  * The Symfony Standard Edition (and therefore, the `symfony new` command) will always give you a clean and empty Symfony application.
  * We'll publish a complete Symfony demo application that combines what we did for the Best Practices book (and which wasn't released) and the current AcmeDemoBundle. This will be a real application using `AppBundle` and all the best practices. No more Acme or Demo.
  * If you are new to Symfony and want to test it ... install the Symfony Installer in 1 minute and execute the `symfony demo` command. In 10 seconds you'll have a full-featured Symfony application ready to test.

And here are some comments regarding this command:

  * For now it doesn't download anything because the `Symfony_Demo` package doesn't exist yet.
  * This command **must** be as simple as possible (remember that it will be used by newcomers that just want to test-drive Symfony quickly and easily). That's why we don't allow to select the Symfony version (it will always be the most recent one). And we also don't allow to select the directory where the demo project is installed (it will be installed in `getcwd().'symfony_demo/'`)

And here it is the `demo` command in action:

symfony_demo_command

### CODE REVIEW

**Comment by Pierstoval** [comment]
:+1: 
 
I'm gonna use this command in my trainings once it becomes available :+1:

**Comment by xabbuh** [comment]
Just a quick suggestion: Instead of copying a lot of code from the `new` command, we should probably some parts to a dedicated layer and reuse it in both commands. As a benefit, this would probably also ease testing them quite a lot (I still need to wrap up my pull request though).

**Comment by javiereguiluz** [comment]
@xabbuh what would you think if I make the `DemoCommand` inherit from the `NewCommand`?

**Comment by xabbuh** [comment]
In this case, an abstract base command class would probably be better.

**Comment by javiereguiluz** [comment]
@xabbuh I've updated this PR to introduce the common abstract method.

File: src/Symfony/Installer/DownloadCommand.php [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+            } else {
+                throw new \RuntimeException(sprintf(
+                    "There was an error downloading %s from symfony.com server:\n%s",
+                    $this->getDownloadedFileName(),
+                    $e->getMessage()
+                ));
+            }
+        }
+
+        $this->fs->dumpFile($this->downloadedFilePath, $response->getBody());
```
Comment (stof) [inline src/Symfony/Installer/DownloadCommand.php]: ``$this->fs`` should be moved to the DownloadCommand as it is used here

File: src/Symfony/Installer/DownloadCommand.php [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+     * available operating system uncompressing commands and the enabled PHP extensions
+     * and it downloads the file.
+     *
+     * @return Command
+     *
+     * @throws \RuntimeException if the Symfony archive could not be downloaded
+     */
+    protected function download()
+    {
+        $this->output->writeln(sprintf("\n Downloading %s...\n", $this->getDownloadedFileName()));
```
Comment (stof) [inline src/Symfony/Installer/DownloadCommand.php]: ``$this->output`` should be moved to this class

Comment (stof) [inline src/Symfony/Installer/DownloadCommand.php]: same for any other property used here btw (look at Insight which can probably warn you IIRC)

File: src/Symfony/Installer/DownloadCommand.php [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+     */
+    protected function getDownloadedFileName()
+    {
+        $commandName = $this->getName();
+
+        if ('new' === $commandName) {
+            return 'Symfony';
+        }
+
+        if ('demo' === $commandName) {
```
Comment (stof) [inline src/Symfony/Installer/DownloadCommand.php]: using an abstract method implemented in each command would be cleaner IMO

**Pull request closed without merging**